kirc.c

```diff
@@ -969,8 +1200,75 @@ static void open_tty()
     }
 }
 
+static void slot_clear(size_t i) {
+    memset(dcc_sessions.slots[i].filename, 0, FNM_MAX);
+    dcc_sessions.sock_fds[i] = (struct pollfd){.fd = -1, .events = POLLIN | POLLOUT};
+    dcc_sessions.slots[i] = (struct dcc_connection){.file_fd = -1};
+}
+
+/* TODO: implicitly assumes that the size reported was the correct size */
+/* TODO: should we just close the file and keep on running if we get
+   an error we can recover from? */
+static void slot_process(state l, char *buf, size_t buf_len, size_t i) {
+    const char *err_str = "";
+    int sock_fd = dcc_sessions.sock_fds[i].fd;
+    int file_fd = dcc_sessions.slots[i].file_fd;
+
+    if (~dcc_sessions.sock_fds[i].revents & POLLIN) {
+        return;
+    }
+
+    int n = read(sock_fd, buf, buf_len);
+    if (n < 0) {
+        err_str = "read";
+        goto handle_err;
+    }
+
+    if (n) {
```

Comment:
For readability, this should be an explicit comparison. i.e. `n > 0`.  this also leaves `n = 0` unchecked.

Reply:
I'll let @emekoi make changes to this and I will merge it to dcc.
The issue I mentioned above should also be resolved.

Reply:
@mcpcpc @emekoi 
This was fixed in my last commits.